### ISSUE TITLE
I ported the parser to TypeScript and used it as a permission DSL

### ISSUE THREAD

**Issue by nicolaes** [gnebehay/parser#8]
If you think it's useful, I can make it into a PR.
https://stackblitz.com/edit/typescript-vhd3tn?file=index.ts

**Comment by gnebehay** [comment]
This is very cool! I'll add a link to the Readme!

**Comment by theRealProHacker** [comment]
For me at least, it is not very clear what a permission DSL is. Searching did not help. Maybe it would be helpful to explain what that means and how it is related to a parser for mathematical expressions.

**Comment by nicolaes** [comment]
Hi @theRealProHacker,

Sorry I'm just now seeing your comment. My code uses the parser to combine permissions.
By permission DSL I understand an AND-OR language that you apply for permissions. Basically a boolean logic parser.
E.g. in Angular:
```html
<div *appDisplayIf="isPremiumUser AND (subscribed OR hasTrial)">
   Premium content.
</div>
```

This boolean logic DSL evolved quite nice since I started using it and is still the most condensed algorithm I found.
It's used by non-technical people in my project to configure UI permissions and display conditions.
